src/script/signingprovider.h

```diff
@@ -74,6 +72,8 @@ class FillableSigningProvider : public SigningProvider
     void ImplicitlyLearnRelatedKeyScripts(const CPubKey& pubkey) EXCLUSIVE_LOCKS_REQUIRED(cs_KeyStore);
 
 public:
+    mutable CCriticalSection cs_KeyStore;
```

Comment:
it seems to be like the reason you made `cs_KeyStore` public is to be able to lock it together with the `spk_man` maybe there could be a `LockScriptPubKey()` function that locks them both instead? 
that way the keystore won't be public and no one will forget to lock the wallet because for the naked eye it looks like the wallet is locked and not used.